### PR TITLE
Rewrite check_urls

### LINKED ISSUE #68: Rewrite check_urls

### PR DESCRIPTION
**Pull request by vkbo**
**Summary**:

This is a rewrite of the `check_urls` function. It now only checks that the url is ASCII, contains one of the following schemes: http, https, ftp, sftp, and checks that it has a domain with at least one dot.

The `full_check` is now enabled again, with the rectangle and url checks, and the vocabulary checks disabled for now.

**Related issue**:

Part of #49, closes #68

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: dmci/mmd_tools/check_mmd.py [hunk 1]
```diff
@@ -174,12 +192,12 @@ def full_check(doc): # pragma: no cover
     urls = [elem.text for elem in url_elements]
     if len(urls) > 0:
         logger.debug('Checking element(s) containing URL ...')
-        urls_ok = check_urls(urls)
+        urls_ok = check_urls(urls, allow_no_path=True)
```
Comment (mortenwh) [inline dmci/mmd_tools/check_mmd.py:195]: Any reason you decided to allow no path?

File: dmci/mmd_tools/check_mmd.py [hunk 2]
```diff
@@ -192,38 +210,40 @@ def full_check(doc): # pragma: no cover
...
-    if voc_ok:
-        logger.info('OK - Controlled vocabularies.')
-    else:
-        logger.info('NOK - Controlled vocabularies -> check debug log')
+    # voc_ok = check_vocabulary(doc)
+    # valid &= voc_ok
+    # if voc_ok:
+    #     logger.info('OK - Controlled vocabularies.')
+    # else:
+    #     logger.info('NOK - Controlled vocabularies -> check debug log')
```
Comment (mortenwh) [inline dmci/mmd_tools/check_mmd.py:247]: why are we not testing this?

**Review by mortenwh — CHANGES_REQUESTED** [review]
Looks good but I have a couple of questions. Based on the tests the url check seems to be as expected, though :)

File: dmci/mmd_tools/check_mmd.py (hunk 2 above)
Comment (vkbo) [reply dmci/mmd_tools/check_mmd.py:247]: Because the vocabulary checks don't work.

File: dmci/mmd_tools/check_mmd.py (hunk 1 above)
Comment (vkbo) [reply dmci/mmd_tools/check_mmd.py:195]: Because the tests failed, but we should maybe check what we want to do here.

**Review by mortenwh — APPROVED** [review]

**Pull request merged**